nailgun/entities.py

```diff
@@ -677,19 +679,27 @@ def create_payload(self):
         For more information, see `Bugzilla #1151220
         <https://bugzilla.redhat.com/show_bug.cgi?id=1151220>`_.
 
+        In addition, rename the ``search_`` field to ``search``.
+
         """
-        return {
-            u'discovery_rule': super(DiscoveryRule, self).create_payload()
-        }
+        payload = super(DiscoveryRule, self).create_payload()
+        if 'search_' in payload:
+            payload['search'] = payload.pop('search_')
+        return {u'discovery_rule': payload}
+
+    def read(self, entity=None, attrs=None, ignore=None):
+        """Rename ``search`` to ``search_``."""
+        if attrs is None:
+            attrs = self.read_json()
+        attrs['search_'] = attrs.pop('search')
```

Comment:
Is it safe to assume that we will always get ``attrs['search']``? Do we need to check if it exists first?